README.md

```diff
@@ -1,4 +1,208 @@
+
+To start receiving current state updates, Calculator UI implementation needs to subscribe to `store` instance:
+```kotlin
+store.subscribe { state -> updateUI(state) }
+```
+By default, when first subscription happens, `store` will emit `initialState` to it.
+
+On each new Calculator UI _intention_, it just need to send it (_dispatch_) as an action to `store` instance:
+```kotlin
+store.dispatch(CalculatorAction.Add(10)) 
+// Current state will become 10
+store.dispatch(CalculatorAction.Deduct(1)) 
+// Current state will become 9
+store.dispatch(CalculatorAction.Add(1)) 
+// Current state will become 10
+store.dispatch(CalculatorAction.Divide(10)) 
+// Current state will become 1
+```
+
+All actions will be processed in a serialized order 
+and on each incoming action `reducer` function will be called to update current state.
+
+### Side effects
+
+Side effect is an interface that has one function - it receives incoming 
+actions, can get current store state at any time and may emit outgoing actions.
+**So basically it is actions in and actions out.** 
+
+**Actions in** to side effects are emitted by store after calling `reducer` function. 
+
+**Actions out** are consumed back again by store and trigger calling `reducer` function.
+
+Basic usage for side effects is to perform some job on incoming action, that may take time (I/O operation) 
+or job that should not update the state (same state should be emitted).
+
+Let's add an side effect that makes a network request each time when action is `CalculatorAction.Add`
+and current state is `9`. If server responds with http code `200` - side effect should emit `CalculatorAction.Deduct(1)`.
+```kotlin
+val sideEffect = object : SideEffect<Int, CalculatorAction> {
+    override val name: String = "network logger"
+
+    override fun CoroutineScope.start(
+        input: ReceiveChannel<CalculatorAction>,
+        stateAccessor: StateAccessor<Int>,
+        output: SendChannel<CalculatorAction>,
+        logger: SideEffectLogger
+    ): Job = launch(context = CoroutineName(name)) {
+        for (inputAction in input) {
+            logger.logSideEffectEvent { LogEvent.SideEffectEvent.InputAction(name, inputAction) }
+            if (inputAction is CalculatorAction.Add &&
+                stateAccessor() == 9) {
+                launch {
+                    val response = makeNetworkCall()
+                    logger.logSideEffectEvent {
+                        LogEvent.SideEffectEvent.Custom(name, "Received network response: $response")
+                    }
+                    if (response == 200) {
+                        val outputAction = CalculatorAction.Deduct(1)
+                        logger.logSideEffectEvent { LogEvent.SideEffectEvent.DispatchingToReducer(name, outputAction) }
+                        output.send(outputAction)
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+Now we need to modify how store is created by adding this `sideEffect`:
```

Comment:
You must register you SideEffect inside `createStore( sideEffects = listOf(...)`